gemfire-core/src/main/java/com/gemstone/gemfire/internal/cache/LocalRegion.java

```diff
@@ -14364,8 +14371,12 @@ private long getEntryOverhead(RegionEntry entry){
 
   protected long calculateEntryOverhead(RegionEntry entry) {
     if (!this.reservedTable() && entryOverHead == -1L && needAccounting()) {
-      entryOverHead = getEntryOverhead(entry);
-      memTrace("Entry overhead for " + getFullPath() + " = " + entryOverHead);
+      synchronized (this){
+        if(entryOverHead == -1L){
+          entryOverHead = getEntryOverhead(entry);
```

Comment:
Please use standard formatting (spaces before braces) etc. Our IDEA settings have it for java already.